File: src/main/java/edu/kit/iti/formal/stvs/model/verification/VerificationResultVisitor.java
package edu.kit.iti.formal.stvs.model.verification;

import java.io.IOException;

/**
 * A visitor for {@link VerificationResult}s.
 */
public interface VerificationResultVisitor {

  /**
   * Visit a {@link Counterexample}.
   * @param result Counterexample to visit.
   */
  void visitCounterexample(Counterexample result);

  /**
   * Visit a {@link VerificationError}.
   * @param result error to visit
   */
  void visitVerificationError(VerificationError result);

  /**
   * Visit a {@link VerificationSuccess}.
   * @param result success to visit
   */
  void visitVerificationSuccess(VerificationSuccess result);
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/model/verification/VerificationResultVisitor.kt
package edu.kit.iti.formal.stvs.model.verification

/**
 * A visitor for [VerificationResult]s.
 */
interface VerificationResultVisitor<T> {
    /**
     * Visit a [Counterexample].
     * @param result Counterexample to visit.
     */
    fun visitCounterexample(result: Counterexample): T

    /**
     * Visit a [VerificationError].
     * @param result error to visit
     */
    fun visitVerificationError(result: VerificationError): T

    /**
     * Visit a [VerificationSuccess].
     * @param result success to visit
     */
    fun visitVerificationSuccess(result: VerificationSuccess): T
    fun visitEmpty(verificationResult: VerificationResult): T
}
